internal/status/multicluster_ing_status.go

```diff
@@ -0,0 +1,124 @@
+// status/annotations.
+func UpdateMultiClusterIngressStatusAndAnnotation(key string, option *UpdateOptions) {
+	ns := option.ServiceMetadata.Namespace
+	ingName := option.ServiceMetadata.IngressName
+	mciObj, err := utils.GetInformers().MultiClusterIngressInformer.Lister().MultiClusterIngresses(ns).Get(ingName)
+	if err != nil {
+		utils.AviLog.Warnf("key: %s, msg: Could not get the Multi-cluster ingress object for update status: %s", key, err)
+		return
+	}
+	mciObj = mciObj.DeepCopy()
+	mciObj.Status.LoadBalancer.Ingress = make([]akov1alpha1.IngressStatus, 1)
+	mciObj.Status.LoadBalancer.Ingress[0].Hostname = mciObj.Spec.Hostname
+	mciObj.Status.LoadBalancer.Ingress[0].IP = option.Vip[0]
+
+	utils.AviLog.Debugf("key: %s, msg: Updating multicluster ingress object status with %+v:", key, utils.Stringify(&mciObj.Status))
+	UpdateMultiClusterIngressStatus(key, mciObj, &mciObj.Status)
+
+	vsAnnotations := make(map[string]string)
+	vsAnnotations[mciObj.Spec.Hostname] = option.VirtualServiceUUID
+
+	if err := patchMultiClusterIngressAnnotations(mciObj, vsAnnotations, key); err != nil {
+		utils.AviLog.Warnf("key: %s, msg: Could not update the Multi-cluster ingress object's annotation: err %s, %s/%s", key, err, mciObj.Namespace, mciObj.Name)
+	}
+}
+
+// UpdateMultiClusterIngressStatus updates MultiClusterIngress' status
+func UpdateMultiClusterIngressStatus(key string, mci *akov1alpha1.MultiClusterIngress, status *akov1alpha1.MultiClusterIngressStatus, retryNum ...int) {
+	retry := 0
+	if len(retryNum) > 0 {
+		retry = retryNum[0]
+		if retry >= 3 {
+			utils.AviLog.Errorf("key: %s, msg: UpdateMultiClusterIngressStatus retried 3 times, aborting", key)
+			return
+		}
+	}
+
+	patchPayload, _ := json.Marshal(map[string]interface{}{
+		"status": status,
+	})
+
+	_, err := lib.AKOControlConfig().CRDClientset().AkoV1alpha1().MultiClusterIngresses(mci.Namespace).Patch(context.TODO(), mci.Name, types.MergePatchType, patchPayload, metav1.PatchOptions{}, "status")
+	if err != nil {
+		utils.AviLog.Errorf("key: %s, msg: there was an error in updating the multicluster ingress status: %+v", key, err)
+		updatedMCI, err := utils.GetInformers().MultiClusterIngressInformer.Lister().MultiClusterIngresses(mci.Namespace).Get(mci.Name)
+		if err != nil {
+			utils.AviLog.Warnf("key: %s, msg: multicluster ingress not found %v", key, err)
+			if strings.Contains(err.Error(), utils.K8S_ETIMEDOUT) {
+				UpdateMultiClusterIngressStatus(key, updatedMCI, status, retry+1)
+			}
+			return
+		}
+		UpdateMultiClusterIngressStatus(key, updatedMCI, status, retry+1)
+		return
+	}
+
+	utils.AviLog.Infof("key: %s, msg: Successfully updated the multicluster ingress %s/%s status %+v", key, mci.Namespace, mci.Name, utils.Stringify(status))
+}
+
+func patchMultiClusterIngressAnnotations(mci *akov1alpha1.MultiClusterIngress, vsAnnotations map[string]string, key string) error {
```

Comment:
Can we rename this to `UpdateMultiClusterIngressAnnotations`? Asking this just for consistency between `UpdateMultiClusterIngressStatus` and `UpdateMultiClusterIngressAnnotations`.

Reply:
I have made the changes.